app/src/main/java/com/example/myapplication/MainActivity.kt

```diff
@@ -229,6 +241,38 @@ class MainActivity : AppCompatActivity() {
         return road
     }
 
+    private fun osrmRouting(
+        startPoint: GeoPoint,
+        endPoint: GeoPoint,
+        mode: String,
+        file: MutableList<RoadNodeForRouting>
+    ) {
+
+        val roadManager: RoadManager = OSRMRoadManager(this, "User")
+        when (mode) {
+            "bike" -> {
+                (roadManager as OSRMRoadManager).setMean(OSRMRoadManager.MEAN_BY_BIKE)
+            }
+
+            "car" -> {
+                (roadManager as OSRMRoadManager).setMean(OSRMRoadManager.MEAN_BY_CAR)
+            }
+
+            else -> {
+                (roadManager as OSRMRoadManager).setMean(OSRMRoadManager.MEAN_BY_FOOT)
+            }
+        }
```

Comment:
```suggestion
        val roadManager = OSRMRoadManager(this, "User")
        when (mode) {
            "bike" -> roadManager.setMean(OSRMRoadManager.MEAN_BY_BIKE)
            "car" -> OSRMRoadManager).setMean(OSRMRoadManager.MEAN_BY_CAR)
            else -> roadManager.setMean(OSRMRoadManager.MEAN_BY_FOOT)
        }
```
+1 pour le swich case